### PR TITLE
Removed presto-geospatial module and moved the types and functions to presto-main

### PR DESCRIPTION
**Pull request by pratyakshsharma**
Relevant slack discussion here - https://prestodb.slack.com/archives/C07JH9WMQ/p1651146423521239

Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* Removed presto-geospatial module 
* Added the types and SqlFunctions defined by GeoPlugin to `BuiltInTypeAndFunctionNamespaceManager.java` for registering
* Moved the relevant classes to presto-main to avoid too many circular dependency issues which would have occurred if movement was done to presto-common rather
* Moved the relevant test classes to presto-main and presto-memory

```

### CODE REVIEW

**Comment by pratyakshsharma** [comment]
@fgwang7w please take a look and add any relevant stakeholders.

cc @imjalpreet @agrawalreetika

File: presto-main/src/main/java/com/facebook/presto/geospatial/SphericalGeoFunctions.java [hunk 1]
```diff
@@ -143,7 +135,7 @@ public static Double stSphericalArea(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) Sli
             return null;
         }
 
-        validateSphericalType("ST_Area", geometry, EnumSet.of(POLYGON, MULTI_POLYGON));
+        validateSphericalType("ST_Area", geometry, EnumSet.of(com.facebook.presto.geospatial.GeometryType.POLYGON, com.facebook.presto.geospatial.GeometryType.MULTI_POLYGON));
```
Comment (fgwang7w) [inline presto-main/src/main/java/com/facebook/presto/geospatial/SphericalGeoFunctions.java:138]: nit: use import static for POLYGON and MULTI_POLYGON members

File: presto-main/src/main/java/com/facebook/presto/geospatial/SphericalGeoFunctions.java [hunk 2]
```diff
@@ -230,7 +222,7 @@ public static Double stSphericalLength(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) S
             return null;
         }
 
-        validateSphericalType("ST_Length", geometry, EnumSet.of(LINE_STRING, MULTI_LINE_STRING));
+        validateSphericalType("ST_Length", geometry, EnumSet.of(com.facebook.presto.geospatial.GeometryType.LINE_STRING, com.facebook.presto.geospatial.GeometryType.MULTI_LINE_STRING));
```
Comment (fgwang7w) [inline presto-main/src/main/java/com/facebook/presto/geospatial/SphericalGeoFunctions.java:225]: nit: use import static instead

File: presto-main/src/main/java/com/facebook/presto/geospatial/SphericalGeoFunctions.java [hunk 3]
```diff
@@ -265,7 +257,7 @@ public static Slice stSphericalCentroid(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME)
             return null;
         }
         // TODO: add support for other types e.g. POLYGON
-        validateSphericalType("ST_Centroid", geometry, EnumSet.of(POINT, MULTI_POINT));
+        validateSphericalType("ST_Centroid", geometry, EnumSet.of(com.facebook.presto.geospatial.GeometryType.POINT, com.facebook.presto.geospatial.GeometryType.MULTI_POINT));
```
Comment (fgwang7w) [inline presto-main/src/main/java/com/facebook/presto/geospatial/SphericalGeoFunctions.java:260]: nit: use import static instead

File: presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java [hunk 4]
```diff
@@ -443,6 +443,11 @@ public void installPlugin(Plugin plugin)
         pluginManager.installPlugin(plugin);
     }
 
+    public PluginManager getPluginManager()
```
Comment (fgwang7w) [inline presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java]: who's the caller of this method? if no caller, please remove this public method

File: presto-main/src/main/java/com/facebook/presto/server/PluginManager.java [hunk 5]
```diff
@@ -212,6 +212,11 @@ private void loadPlugin(URLClassLoader pluginClassLoader)
         }
     }
 
+    public Metadata getMetadata()
```
Comment (fgwang7w) [inline presto-main/src/main/java/com/facebook/presto/server/PluginManager.java]: who's the caller of this method? if no caller, pls remove this public method

File: presto-main/src/test/java/com/facebook/presto/geospatial/TestExtractSpatialInnerJoin.java [hunk 6]
```diff
@@ -46,7 +46,7 @@
 
     public TestExtractSpatialInnerJoin()
     {
-        super(new GeoPlugin());
+//        super(new GeoPlugin());
```
Comment (fgwang7w) [inline presto-main/src/test/java/com/facebook/presto/geospatial/TestExtractSpatialInnerJoin.java]: nit: remove this line

File: presto-main/src/test/java/com/facebook/presto/geospatial/TestExtractSpatialLeftJoin.java [hunk 7]
```diff
@@ -45,7 +45,7 @@
 
     public TestExtractSpatialLeftJoin()
     {
-        super(new GeoPlugin());
+//        super(new GeoPlugin());
```
Comment (fgwang7w) [inline presto-main/src/test/java/com/facebook/presto/geospatial/TestExtractSpatialLeftJoin.java]: nit: remove this line

File: presto-main/src/test/java/com/facebook/presto/geospatial/TestRewriteSpatialPartitioningAggregation.java [hunk 8]
```diff
@@ -34,7 +34,7 @@
 {
     public TestRewriteSpatialPartitioningAggregation()
     {
-        super(new GeoPlugin());
+//        super(new GeoPlugin());
```
Comment (fgwang7w) [inline presto-main/src/test/java/com/facebook/presto/geospatial/TestRewriteSpatialPartitioningAggregation.java]: nit: remove this line

File: presto-memory/src/test/java/com/facebook/presto/plugin/memory/geospatial/BenchmarkGeometryAggregations.java [hunk 9]
```diff
@@ -69,7 +69,7 @@ public void setUp()
                     .setCatalog("memory")
                     .setSchema("default")
                     .build());
-            queryRunner.installPlugin(new GeoPlugin());
+//            queryRunner.installPlugin(new GeoPlugin());
```
Comment (fgwang7w) [inline presto-memory/src/test/java/com/facebook/presto/plugin/memory/geospatial/BenchmarkGeometryAggregations.java]: nit: remove this line

File: presto-memory/src/test/java/com/facebook/presto/plugin/memory/geospatial/BenchmarkSpatialJoin.java [hunk 10]
```diff
@@ -81,7 +81,7 @@ public void setUp()
                     .setCatalog("memory")
                     .setSchema("default")
                     .build());
-            queryRunner.installPlugin(new GeoPlugin());
+//            queryRunner.installPlugin(new GeoPlugin());
```
Comment (fgwang7w) [inline presto-memory/src/test/java/com/facebook/presto/plugin/memory/geospatial/BenchmarkSpatialJoin.java]: nit: remove this line

File: presto-memory/src/test/java/com/facebook/presto/plugin/memory/geospatial/TestSpatialJoinPlanning.java [hunk 11]
```diff
@@ -72,7 +72,7 @@ private static LocalQueryRunner createQueryRunner()
                 .setCatalog("memory")
                 .setSchema("default")
                 .build());
-        queryRunner.installPlugin(new GeoPlugin());
+//        queryRunner.installPlugin(new GeoPlugin());
```
Comment (fgwang7w) [inline presto-memory/src/test/java/com/facebook/presto/plugin/memory/geospatial/TestSpatialJoinPlanning.java]: nit: remove this line

**Review by fgwang7w — CHANGES_REQUESTED** [review]
Thank you for making the changes. Please merge all 4 commits into one. GeoQueryRunner is removed in this PR so it'd be better to include this note in a commit message to document that geospatial function is supported in mainline logic and does not need to include any geoplugin to enable the feature.

**Comment by fgwang7w** [comment]
Hi @zhenxiao I've given 1st pass of code review, please help review as well. 
Hi @tdcmeehan This is a straight forward lib moving from one module to `presto-main`, I suppose there's a need to document the changes for release notes, any suggestion? Thanks

File: presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java [hunk 12]
```diff
@@ -683,6 +704,12 @@ private List<? extends SqlFunction> getBuildInFunctions(FeaturesConfig featuresC
                 .scalars(JsonFunctions.class)
                 .scalars(ColorFunctions.class)
                 .scalars(ColorOperators.class)
+                .scalars(GeoFunctions.class)
+                .scalars(BingTileFunctions.class)
+                .scalars(BingTileOperators.class)
+                .scalar(BingTileFunctions.BingTileCoordinatesFunction.class)
+                .scalars(SphericalGeoFunctions.class)
+                .scalars(KdbTreeCasts.class) //TODO: check this
```
[the remark this answers is not in the document]
Comment (pratyakshsharma) [reply presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java:712]: removed

**Review by tdcmeehan — APPROVED** [review]

**Comment by tdcmeehan** [comment]
@pratyakshsharma please fix the merge conflicts, thanks!

**Comment by pratyakshsharma** [comment]
fixed the merge conflicts. Verified the changes using `show functions;` command and by invoking few simple functions as in the attached screenshot
Screenshot 2022-08-17 at 10 11 45 PM
.

**Comment by pratyakshsharma** [comment]
@tdcmeehan let us merge this!

**Comment by pratyakshsharma** [comment]
@arunthirupathi sure. Actually the user facing release notes are present in the commit message body. Providing them as below - 

1. presto-geospatial module has been removed and GeoPlugin class has been deleted. 
2. User does not need to do anything now to make use of presto's geospatial functions. All the functions are auto registered, one can simply start using them after starting the server. 

Does this look good?

**Pull request merged**